src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -214,6 +219,26 @@ private ValueTask<HttpConnection> GetConnectionAsync(HttpRequestMessage request,
                 new HttpConnectionWithFinalizer(this, stream, transportContext); // finalizer needed to signal the pool when a connection is dropped
         }
 
+        public async ValueTask<HttpConnection> UpgradeConnectionToTls(HttpRequestMessage request,  Stream stream, CancellationToken cancellationToken)
+        {
+            SslStream sslStream = await ConnectHelper.EstablishSslConnectionAsync(_sslOptions, request, stream, cancellationToken).ConfigureAwait(false);
+            HttpConnection conn = new HttpConnection(this, sslStream, sslStream.TransportContext);
+
+            if (NetEventSource.IsEnabled)
+            {
+                conn.Trace(
```

Comment:
It is protected with the proper `NetEventSource.IsEnabled` and shouldn't happen that frequently but due to https://github.com/dotnet/csharplang/issues/939 it is going to generate a bunch of `string.Format` calls and respective allocations when tracing is enabled.